server/routes/api/choropleth.py

```diff
@@ -32,76 +32,92 @@
-        mdom -> The measurement denominator to use if perCapita is true, as a 
-                string. Defaults to "Count_Person".
-
+        statVar: The statistical variable to pull data about.
 
     API Returns:
-        df -> Returns a dictionary with two entries
-            geoJson -> geoJson format that includes statistical variables info,
-                    geoId, and name for all subregions.
-            _PLOTTING_INFO -> Includes domain range for use in D3 and the
-                    color palette to render with.
+        values: dictionary of geo to statistical variable values (as a float)
+            for all subgeos.
     """
     # Get required request parameters.
-    requested_stat_var = request.args.get("statVar")
-    if requested_stat_var is None:
+    requested_geoDcid = request.args.get("geoDcid")
+    if not requested_geoDcid:
+        return jsonify({"error": "Must provide a 'geoDcid' field!"}, 400)
+    stat_var = request.args.get("statVar")
+    if not stat_var:
         return jsonify({"error": "Must provide a 'statVar' field!"}, 400)
-    requested_geoId = request.args.get("geoId")
-    if requested_geoId is None:
-        return jsonify({"error": "Must provide a 'geoId' field!"}, 400)
-
-    # If no level is requested then default to one level below current.
-    display_level = request.args.get("level")
-    if display_level is None:
-        requested_geoId_type = dc.get_property_values([requested_geoId],
-                                                "typeOf")[requested_geoId]
-        # TODO(iancostello): Handle a failed function call, e.g., returns None.
-        # TODO(iancostello): Handle the case where display_level is None.
-        for level in requested_geoId_type:
-            if level in LEVEL_MAP:
-                display_level = LEVEL_MAP[level]
-                break
+    display_level = get_sublevel(requested_geoDcid, request.args.get("level"))
+
+    # Get all subgeos.
+    geos_contained_in_place = dc.get_places_in(
+            [requested_geoDcid], display_level).get(requested_geoDcid, [])
+    values_by_geo = dc.get_stats(geos_contained_in_place, stat_var)
+
+    # Add to dictionary for response.
+    populations_by_geo = {}
+    for geo_id, payload in values_by_geo.items():
+        if "data" in payload:
+            populations_by_geo[geo_id] = next(iter(
+                        reversed(payload['data'].values())))
+
+    # Return as json payload.
+    return jsonify(populations_by_geo, 200)
+
+@bp.route('/geo')
+def choropleth_geo():
+    """Returns data for geographic subregions for a certain statistical 
+            variable.
+
+    API Params:
+        geoDcid: The currently viewed geography to render, as a string.
+        level: The subgeographic level to pull and display information for,
+                as a string. Choices: Country, AdministrativeLevel[1/2], City.
+        mdom: The measurement denominator to use as a string.
```

Comment:
I'm not sure how the user will end up selecting a separate denominator for per capita computations. Elsewhere, we simply default to Count_Person.

Reply:
It's not implemented in the UI currently, but its possible via adding an attribute to the UI.

Reply:
I understand, but I don't see us adding this as a UI feature (so we shouldn't complicate the code for an unsupported feature).